### DISCUSSION TITLE
what is payload maxium size? I want to store long document

### DISCUSSION THREAD

**Discussion by hiepxanh** [qdrant/qdrant#3934] [General]
about 1 or 2 A4 page. What is maximum payload size? I cannot find in the docs. Currently I use qdrant to search and get the document_id then go to postgres to take full document, it take 0s to qdrant and 3 second to postgres, what if I put postgres document to qdrant payload? how much it can handle and does it effect to search speed or not? 
I assume that payload is load from disk, ram store index so it will be not problem

**Comment by timvisee** [comment]
There is no hard limit. In the end it will be limited by the hardware you provide.

In a collection, you can set the `payload_on_disk: true` flag to ensure all payload remains on disk ([docs](https://qdrant.tech/documentation/concepts/collections/#create-a-collection)).

> I assume that payload is load from disk, ram store index so it will be not problem

It is if you configure it to be on disk.

> how much it can handle and does it effect to search speed or not?

Generally it should not. But if your payload is huge, and your disks are slow, actually retrieving the search results (after the search process) may take a bit longer.

I'd recommend to give it a test to see whether it satisfies your needs.

**Reply by hiepxanh** [reply]
It was a great answer, I love this, the docs is clear and the power is unlimit 😍😍😍😍
Thank you
